sphinxcontrib/bitfield.py

```diff
@@ -1,65 +1,37 @@
 from docutils import nodes
-from docutils.parsers.rst import Directive
+from docutils.parsers.rst import Directive, directives
 from docutils.parsers.rst.directives import flag
 from bit_field import render, jsonml_stringify
 from json import loads
-from hashlib import sha1
-from os.path import join
-from sphinx.errors import ExtensionError
 from shlex import split
-
-
-class bitfield(nodes.General, nodes.Element):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.options = dict(kwargs['options'])
-        self.options['compact'] = 'compact' in self.options
```

Comment:
Nice. overall a lot simpler.

Only issue I had was related to removing these lines. The `flag` conversion function used for these options sets the value to None if the option is present.  ie: `self.options["compact"] == None` when `:compact:` is in the directive. `bit_field.render` naturally considers those options `False`. 

Tacked on a commit with a conversion function that works the same as `flag` but will set the value to `True` when present instead of `None`. After a quick scan of `docutils.directives` I didnt see a conversion function like this.

Otherwise everything worked as expected

Reply:
rebased onto latest master

Reply:
Good catch, thanks!